### ISSUE TITLE
Error adding Ragnar to a IIS Website

### ISSUE THREAD

**Issue by ngjermundshaug** [libragnar/ragnar#100]
Hi

I am trying to include Ragnar into a IIS Website. After adding Nuget package and running the site (.Net 4.5.1 - x86) I get this error: 

![image](https://cloud.githubusercontent.com/assets/1677884/9196133/893368ae-4028-11e5-9149-92955a044b5c.png)

the **/bin** folder contains these files: 

![image](https://cloud.githubusercontent.com/assets/1677884/9196146/abb72528-4028-11e5-89fa-1051e7bb408a.png)

Have also tried 
* to compile and use my own DLL files - same thing happens
* allow unsafe code in project settings
* adding assemblyBinding config section in web.config
* renamed Ragnar.dll to Ragar.DLL in the bin folder
* reinstalled x64 and x86 C++ 2013 Redistributables
* Changed OutputType of RagnarClient from **Application** to **Class Library** & recompiled
* Tried running IISExpress and also on IIS 7.5 (Win 2008 R2) - same error

I have a feeling that I'm missing something obvious here - what am I doing wrong? :)

**Comment by diantahoc** [comment]
What you are trying to do basically is to load a managed C++ library. 

I have searched a few terms and found what might be a solution, check out this answer on StackOverflow: http://stackoverflow.com/a/345032

Ragner.dll might be loaded but it cannot find the torrent.dll file. Maybe try putting the torrent.dll file and it's boost dependencies in `\System32\Inetsrv`? If that doesn't work then I have no other idea.

On a side note: If you are developing a website that download torrents, then opening a ragnar/libtorrent session inside an asp.net session doesn't seem the right way to do it. A better approach is to develop a localhost server / Windows service and use asp.net to communicate with? 

Hadouken (https://github.com/hadouken/hadouken) is a windows service that use Ragnar to download torrents. You might fork it and provide an API (JSON?) and use that to communicate between your asp.net and the service.

Good luck.

**Comment by vktr** [comment]
Hi,

If what @diantahoc is suggesting doesn't work, can you check the bitness of the application pool? Ragnar is purely x86 (32-bit) and a 64-bit application pool might have problems loading the dll's.

Also, I agree with @diantahoc in that Ragnar might not work very well at all in an IIS application. What are you trying to do?

**Comment by ngjermundshaug** [comment]
I ran it as 32-bit. Should theoretically work OK in IIS as long as recycle settings in the IIS App Pool is set properly. I was basically looking for a Torrent Server with HTTP support for communicating with the server - so I thought - what the hell - let's try to load it into IIS and see how it works out.

I also tried to use Hadoken, but it had very limited API for the JS Plugins. @vktr If the JS plugins of Hadouken could support Node.js - it would be awesome. I was missing functionality to browse directories, do HTTP communication etc.

I am now implementing it as a TopShelf Windows Service with a HTTPListener - for communication with the service. Seems to work well. I also got it to seed very fast after disabling UTP.

    settings.EnableOutgoingUtp = false;
